### PR TITLE
Fix AppVeyor 3.4 failing builds

### LINKED ISSUE #227: AppVeyor 3.4 environment failures during scipy import
This is now happening consistently.  E.g. here: https://ci.appveyor.com/project/spencerahill/aospy/build/1.0.358/job/gcafed5igbbtebgt

Googling the error message led me to [this](https://github.com/scikit-image/scikit-image/issues/2207), which may be relevant, but I haven't actually looked into it enough to know for sure.

### PR DESCRIPTION
**Pull request by spencerahill**
Closes #227 

My approach here is basically just guess-and-check until I find something that works, following solutions other packages have found from the link in #227 and within.

### CODE REVIEW

**Comment by spencerkclark** [comment]
@spencerahill rather than what you've tried here, as a temporary workaround, could you try simply pinning `scipy` to version `0.19.0` in our `ci/environment-py34.yml` file?  This is the version before the version that is being installed now. My intuition is that this is an upstream packaging issue (since things were working fine before, but recently stopped working).

File: ci/environment-py34.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ channels:
   - conda-forge
 dependencies:
   - python=3.4
-  - scipy
+  - scipy=0.19
```
Comment (spencerkclark) [inline ci/environment-py34.yml]: I think you may need to pin `0.19.0` explicitly (it seems `0.19.1` is being installed): https://travis-ci.org/spencerahill/aospy/jobs/301587193#L641

Comment (spencerahill) [reply ci/environment-py34.yml]: Blarg.  Good catch, thanks.

**Comment by spencerahill** [comment]
(The most recent commit title incorrectly says 0.19.1 but it is in fact being pinned to 0.19.0)

**Comment by spencerkclark** [comment]
From https://travis-ci.org/spencerahill/aospy/jobs/301591417#L567:
```
UnsatisfiableError: The following specifications were found to be in conflict:
  - python 3.4*
  - scipy 0.19.0* -> python 2.7*
Use "conda info <package>" to see the dependencies for each package.
```
Oops, sorry!  Maybe as one last try, go with `scipy=0.18`?

Some Python 3.4 builds do appear when I run `$ conda info scipy=0.18`:
<details>
```
$ conda info scipy=0.18
Fetching package metadata .........

scipy 0.18.0 np111py27_0
------------------------
file name   : scipy-0.18.0-np111py27_0.tar.bz2
name        : scipy
version     : 0.18.0
build string: np111py27_0
build number: 0
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2016-07-25
license     : BSD
md5         : 7dc0a04708a241a7f7fbe281ca604499
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.0-np111py27_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 2.7*

scipy 0.18.0 np111py27_nomkl_0
------------------------------
file name   : scipy-0.18.0-np111py27_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.0
build string: np111py27_nomkl_0
build number: 0
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2016-07-25
features    : nomkl
license     : BSD
md5         : 4c6fb0c7a7cd4df746d2813268307c4c
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.0-np111py27_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 2.7*

scipy 0.18.0 np111py34_0
------------------------
file name   : scipy-0.18.0-np111py34_0.tar.bz2
name        : scipy
version     : 0.18.0
build string: np111py34_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-07-25
license     : BSD
md5         : 1613c596d34fae4f10b2083c53180a42
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.0-np111py34_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 3.4*

scipy 0.18.0 np111py34_nomkl_0
------------------------------
file name   : scipy-0.18.0-np111py34_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.0
build string: np111py34_nomkl_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-07-25
features    : nomkl
license     : BSD
md5         : 421b2bce2af020143b93eb749654f8b1
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.0-np111py34_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 3.4*

scipy 0.18.0 np111py35_0
------------------------
file name   : scipy-0.18.0-np111py35_0.tar.bz2
name        : scipy
version     : 0.18.0
build string: np111py35_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-07-25
license     : BSD
md5         : f9029b71793ed80ec821e3193e175e61
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.0-np111py35_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 3.5*

scipy 0.18.0 np111py35_nomkl_0
------------------------------
file name   : scipy-0.18.0-np111py35_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.0
build string: np111py35_nomkl_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-07-25
features    : nomkl
license     : BSD
md5         : fe91159c4007de75cc22e1e91041715e
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.0-np111py35_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 3.5*

scipy 0.18.1 np111py27_0
------------------------
file name   : scipy-0.18.1-np111py27_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py27_0
build number: 0
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2016-09-19
license     : BSD
md5         : 83bdc9a63ddfb3055b100ee35cfedfc7
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py27_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 2.7*

scipy 0.18.1 np111py27_nomkl_0
------------------------------
file name   : scipy-0.18.1-np111py27_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py27_nomkl_0
build number: 0
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2016-09-19
features    : nomkl
license     : BSD
md5         : 7b5a123179c51f1f6ca7a98be023da8b
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py27_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 2.7*

scipy 0.18.1 np111py34_0
------------------------
file name   : scipy-0.18.1-np111py34_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py34_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-09-19
license     : BSD
md5         : bd2d9b13c269f8ec0204a76d1c004650
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py34_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 3.4*

scipy 0.18.1 np111py34_nomkl_0
------------------------------
file name   : scipy-0.18.1-np111py34_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py34_nomkl_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-09-19
features    : nomkl
license     : BSD
md5         : 69bc53d3b2e94a6a70cf4cd749283803
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py34_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 3.4*

scipy 0.18.1 np111py35_0
------------------------
file name   : scipy-0.18.1-np111py35_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py35_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-09-19
license     : BSD
md5         : 65f504988c05ae466c93d3d7db6b2560
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py35_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 3.5*

scipy 0.18.1 np111py35_nomkl_0
------------------------------
file name   : scipy-0.18.1-np111py35_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py35_nomkl_0
build number: 0
channel     : defaults
size        : 12.3 MB
arch        : x86_64
date        : 2016-09-19
features    : nomkl
license     : BSD
md5         : 0f85eaa0bad2a994dd68d541cb7351f5
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py35_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 3.5*

scipy 0.18.1 np111py36_0
------------------------
file name   : scipy-0.18.1-np111py36_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py36_0
build number: 0
channel     : defaults
size        : 13.5 MB
arch        : x86_64
date        : 2016-12-23
license     : BSD
md5         : 1258c396aeee16ebef488ae85b21513b
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py36_0.tar.bz2
dependencies:
    mkl 11.3.3
    numpy 1.11*
    python 3.6*

scipy 0.18.1 np111py36_nomkl_0
------------------------------
file name   : scipy-0.18.1-np111py36_nomkl_0.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py36_nomkl_0
build number: 0
channel     : defaults
size        : 13.5 MB
arch        : x86_64
date        : 2016-12-30
features    : nomkl
license     : BSD
md5         : 66dda522e5476d6b73e3d7b6f36eaac9
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py36_nomkl_0.tar.bz2
dependencies:
    numpy 1.11*
    python 3.6*

scipy 0.18.1 np111py27_1
------------------------
file name   : scipy-0.18.1-np111py27_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py27_1
build number: 1
channel     : defaults
size        : 13.8 MB
arch        : x86_64
date        : 2017-01-10
license     : BSD
md5         : ba23893454a9da289d12760becd1c67b
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py27_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.11*
    python 2.7*

scipy 0.18.1 np111py27_nomkl_1
------------------------------
file name   : scipy-0.18.1-np111py27_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py27_nomkl_1
build number: 1
channel     : defaults
size        : 13.7 MB
arch        : x86_64
date        : 2017-01-10
features    : nomkl
license     : BSD
md5         : 111a5ce04319a05d7d6daa5bcadf7c60
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py27_nomkl_1.tar.bz2
dependencies:
    numpy 1.11*
    python 2.7*

scipy 0.18.1 np111py34_1
------------------------
file name   : scipy-0.18.1-np111py34_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py34_1
build number: 1
channel     : defaults
size        : 13.3 MB
arch        : x86_64
date        : 2017-01-10
license     : BSD
md5         : 3ae2f9bb03acb483c51e91089c875ca6
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py34_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.11*
    python 3.4*

scipy 0.18.1 np111py34_nomkl_1
------------------------------
file name   : scipy-0.18.1-np111py34_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py34_nomkl_1
build number: 1
channel     : defaults
size        : 13.5 MB
arch        : x86_64
date        : 2017-01-10
features    : nomkl
license     : BSD
md5         : 9aa496abf5b8d3e7f52f0a2564a06cf7
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py34_nomkl_1.tar.bz2
dependencies:
    numpy 1.11*
    python 3.4*

scipy 0.18.1 np111py35_1
------------------------
file name   : scipy-0.18.1-np111py35_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py35_1
build number: 1
channel     : defaults
size        : 13.3 MB
arch        : x86_64
date        : 2017-01-10
license     : BSD
md5         : 1c868273fa47045978c6c439b238bccf
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py35_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.11*
    python 3.5*

scipy 0.18.1 np111py35_nomkl_1
------------------------------
file name   : scipy-0.18.1-np111py35_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py35_nomkl_1
build number: 1
channel     : defaults
size        : 13.5 MB
arch        : x86_64
date        : 2017-01-10
features    : nomkl
license     : BSD
md5         : 0a975260d78ac61a40b8d56205a8b8b2
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py35_nomkl_1.tar.bz2
dependencies:
    numpy 1.11*
    python 3.5*

scipy 0.18.1 np111py36_1
------------------------
file name   : scipy-0.18.1-np111py36_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py36_1
build number: 1
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2017-01-10
license     : BSD
md5         : efe357a7dbda06f92d26865cfade1ef1
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py36_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.11*
    python 3.6*

scipy 0.18.1 np111py36_nomkl_1
------------------------------
file name   : scipy-0.18.1-np111py36_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np111py36_nomkl_1
build number: 1
channel     : defaults
size        : 13.5 MB
arch        : x86_64
date        : 2017-01-10
features    : nomkl
license     : BSD
md5         : 7b7a0d41cabafaa5f7683dc9aaa3ef7d
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np111py36_nomkl_1.tar.bz2
dependencies:
    numpy 1.11*
    python 3.6*

scipy 0.18.1 np112py27_1
------------------------
file name   : scipy-0.18.1-np112py27_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np112py27_1
build number: 1
channel     : defaults
size        : 13.8 MB
arch        : x86_64
date        : 2017-02-03
license     : BSD
md5         : a7c19a8108d73966d1aa67a35117fcb3
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np112py27_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.12*
    python 2.7*

scipy 0.18.1 np112py27_nomkl_1
------------------------------
file name   : scipy-0.18.1-np112py27_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np112py27_nomkl_1
build number: 1
channel     : defaults
size        : 13.7 MB
arch        : x86_64
date        : 2017-02-04
features    : nomkl
license     : BSD
md5         : c0a4a792ed0df19826904527df72fa50
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np112py27_nomkl_1.tar.bz2
dependencies:
    numpy 1.12*
    python 2.7*

scipy 0.18.1 np112py35_1
------------------------
file name   : scipy-0.18.1-np112py35_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np112py35_1
build number: 1
channel     : defaults
size        : 13.3 MB
arch        : x86_64
date        : 2017-02-04
license     : BSD
md5         : 15e476c7f24f09edf50055a636b41e26
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np112py35_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.12*
    python 3.5*

scipy 0.18.1 np112py35_nomkl_1
------------------------------
file name   : scipy-0.18.1-np112py35_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np112py35_nomkl_1
build number: 1
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2017-02-06
features    : nomkl
license     : BSD
md5         : 99c8c1613ee557e70960a04956543090
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np112py35_nomkl_1.tar.bz2
dependencies:
    numpy 1.12*
    python 3.5*

scipy 0.18.1 np112py36_1
------------------------
file name   : scipy-0.18.1-np112py36_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np112py36_1
build number: 1
channel     : defaults
size        : 13.4 MB
arch        : x86_64
date        : 2017-02-04
license     : BSD
md5         : 29da2cf5e1bf46d502c6a036b0610f7f
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np112py36_1.tar.bz2
dependencies:
    mkl 2017.0.*
    numpy 1.12*
    python 3.6*

scipy 0.18.1 np112py36_nomkl_1
------------------------------
file name   : scipy-0.18.1-np112py36_nomkl_1.tar.bz2
name        : scipy
version     : 0.18.1
build string: np112py36_nomkl_1
build number: 1
channel     : defaults
size        : 13.5 MB
arch        : x86_64
date        : 2017-02-04
features    : nomkl
license     : BSD
md5         : 5de3d5f6c932716f74d30582d58a8db8
noarch      : None
platform    : darwin
url         : https://repo.continuum.io/pkgs/free/osx-64/scipy-0.18.1-np112py36_nomkl_1.tar.bz2
dependencies:
    numpy 1.12*
    python 3.6*
```
</details>

**Comment by spencerkclark** [comment]
Hmm...my intuition was wrong.  As a result of the latest failed build, I went back and looked at the last Python 3.4 build that passed on AppVeyor, and it used scipy version `0.19.1`, so that doesn't appear to be the issue:
https://ci.appveyor.com/project/spencerahill/aospy/build/1.0.342/job/kehc41p848kj984y#L85

Another clue from looking at the build history: the change (it must be upstream, right?) happened sometime between 22 and 15 days ago.

**Comment by spencerahill** [comment]
> 0.19.1, so that doesn't appear to be the issue:

Ok, so that's one less thing 😄 

Will keep looking and trying different things.

**Comment by spencerahill** [comment]
Here is essentially the exact same problem of the `_ufuncs` import from `scipy.special`: https://github.com/numpy/numpy/issues/9933.  Looks like they fixed it by changing the 'mkl' version.  Will try that.

**Comment by spencerkclark** [comment]
Great find, the timeframe and issue seem to align quite well with what we're seeing.  Go for it!

**Comment by spencerahill** [comment]
They specified the scipy, numpy, and mkl versions, but I'm starting with just the MKL version to see if that does the trick.  If not then I'll add the other two.

**Comment by spencerahill** [comment]
Same error: https://ci.appveyor.com/project/spencerahill/aospy/build/1.0.374/job/tgkceenwijg5debr#L485

So pinning scipy as well.

**Comment by spencerahill** [comment]
The way our `appveyor.yml` is set up, the dependencies all get read in from the `environment-py34.yml` file, such that adding the `CONDA_DEPENDENCIES` option within `appveyor.yml` doesn't do anything.  We would have to more closely emulate [stingray's appveyor.yml](https://github.com/StingraySoftware/stingray/blob/2743012a655249c38472ef557e534858a431f0b6/appveyor.yml) for it to work.

So as an alternative I'm trying pinning MKL and scipy within the environment file.

**Comment by spencerahill** [comment]
Phew, looks like that did the trick: https://ci.appveyor.com/project/spencerahill/aospy/build/1.0.378/job/fkhaa22tnqbkt2w1

The downside of making the change within the environment file rather than `appveyor.yml` is that it also affects the Travis build.  But at this point I'm willing to live with this and just be done with this.  We can revisit as necessary if anybody starts reporting bugs using 3.4.

Will merge once the builds finish.

**Comment by spencerkclark** [comment]
Nice, thanks for sorting this out!  If we really wanted to, we *could* separate the Appveyor and Travis build environments, but it’s probably not worth it.

**Pull request merged**